design/resource-constraints.md

```diff
@@ -0,0 +1,52 @@
+# Resource Constraints
+
+## Overview
+Resource Constraints allow the rook components to be in specific Kubernetes Quality of Service (QoS) classes. For this the components started by rook need to have resource requests and/or limits set depending on which class the component(s) should be in.
+
+Ceph has recommendations for CPU and memory for each component. The recommendations can be found here: http://docs.ceph.com/docs/master/start/hardware-recommendations/
+
+## Application of Resource Constraints
+The resource constraints are defined in the rook Cluster, Filesystem and RGW CRDs.
+
+### User defined
+User defined values, when given, are above the calculated values.
+
+#### Global User Resource Specification
+A Kubernetes resource requirement object looks like this:
+```yaml
+requests:
+  cpu: “2”
+  memory: “1Gi”
+limits:
+  cpu: “3”
+  memory: “2Gi”
+```
+
+The key in the CRDs to set resource requirements is named `resources`.
+The following components will allow a resource requirement to be set for:
+* `api`
+* `mds`
+* `mgr`
+* `mon`
+* `osd`
+* `rgw`
+
+#### Special Case: OSD
+To allow per node/OSD configuration of resource constraints, a key is added to the `storage.nodes` item.
+It is named `resources` and contain a resource requirement object (see above).
+
+### Automatic Algorithm
+The user is able to enable and disable the automatic resource algorithm as he wants.
+
+This algorithm has a global "governance" class for the Quality of Service (QoS) class to be aimed for.
+The key to choose the QoS class is named `qualityClass` and in the `resources` specification.
```

Comment:
should we match the k8s property called `qosClass`?

Reply:
Yes sounds better. Fixed.